File: Februrary 2025/27.cpp
#include <iostream>
#include <stack>

class Solution {
    std::stack<int> stack;
    std::stack<int> min_stack;

  public:
    Solution() {
        // Constructor initializes the stacks
    }

    // Add an element to the top of Stack
    void push(int x) {
        stack.push(x);
        if (min_stack.empty() || x <= min_stack.top()) {
            min_stack.push(x);
        }
    }

    // Remove the top element from the Stack
    void pop() {
        if (!stack.empty()) {
            if (stack.top() == min_stack.top()) {
                min_stack.pop();
            }
            stack.pop();
        }
    }

    // Returns top element of the Stack
    int peek() {
        if (!stack.empty()) {
            return stack.top();
        }
        return -1; // or throw an exception if appropriate
    }

    // Finds minimum element of Stack
    int getMin() {
        if (!min_stack.empty()) {
            return min_stack.top();
        }
        return -1; // or throw an exception if appropriate
    }
};

File: Februrary 2025/27.py
class Solution:

    def __init__(self):
        self.stack = []
        self.min_stack = []

    # Add an element to the top of Stack
    def push(self, x):
        self.stack.append(x)
        if not self.min_stack or x <= self.min_stack[-1]:
            self.min_stack.append(x)

    # Remove the top element from the Stack
    def pop(self):
        if self.stack:
            top = self.stack.pop()
            if top == self.min_stack[-1]:
                self.min_stack.pop()

    # Returns top element of Stack
    def peek(self):
        if self.stack:
            return self.stack[-1]
        return -1

    # Finds minimum element of Stack
    def getMin(self):
        if self.min_stack:
            return self.min_stack[-1]
        return -1
